File: c/gpt/p124_nr_ver1.c
/* C */
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

static void rtrim_inplace(char *s) {
    size_t len = strlen(s);
    while (len > 0 && isspace((unsigned char)s[len - 1])) {
        len--;
    }
    s[len] = '\0';
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        fprintf(stderr, "Error: expected exactly one argument.\n");
        return 2;
    }

    const char *arg = argv[1];
    printf("%s\n", arg);

    size_t n = strlen(arg);
    char *input = (char *)malloc(n + 1);
    if (!input) {
        fprintf(stderr, "Error: out of memory.\n");
        return 1;
    }
    memcpy(input, arg, n + 1);

    rtrim_inplace(input);
    printf("%s\n", input);

    free(input);
    return 0;
}

File: python/gpt/p124_nr_ver1.py
# Python
import sys

def main() -> int:
    if len(sys.argv) != 2:
        print("Error: expected exactly one argument.", file=sys.stderr)
        return 2

    input_str = sys.argv[1]
    print(input_str)

    input_str = input_str.rstrip()
    print(input_str)

    return 0

if __name__ == "__main__":
    raise SystemExit(main())